src/mode/modeHandler.ts

```diff
@@ -186,7 +186,7 @@ export class VimState {
   public cursorPositionJustBeforeAnythingHappened = [ new Position(0, 0) ];
 
   public isRecordingMacro: boolean = false;
-  public isReplayingMacro: boolean = false;
+  public isReplayingMacro: number = 0;
```

Comment:
Can you rename this variable to something along the lines of `replayingMacroCount`